### DISCUSSION TITLE
Problem when running native cmd lines in Windows PowerShell

### DISCUSSION THREAD

**Discussion by Siudya** [xmake-io/xmake#5709] [Q&A] [answered]
I installed xmake 2.9.5 using scoop via `scoop install xmake`
And I wrote an `xmake.lua` as follow:
```
target("test")
  set_kind("phony")
  on_run(function (tgt)
    os.exec("Write-Output \"aaaa\"")
  end)
target_end()
```
I run `xmake run test` but it print
```
 error: cannot exec(Write-Output "aaaa"), No such file or directory
```
I try to run the `Write-Output "aaaa"` in powershell and it works.
Is there anything more I should do before running a powershell cmd via xmake?

**Comment by waruqi** [comment]
Write-Output is not window executable program. it's powershell's command. you should run `pwsh` to call these commands

**Reply by Siudya** [reply]
I am trying to use `mill` on both Windows and Linux (I used scoop to install `mill` on windows).  I have to write like this
```
target("test")
  set_kind("phony")
  on_run(function (tgt)
    os.execv(os.shell(), {"mill",  "version"})
  end)
target_end()
```
And I cannot use the exec API on Windows:
```
target("test")
  set_kind("phony")
  on_run(function (tgt)
    os.exec("mill version")
  end)
target_end()
```
It's a little confusing. Maybe it requires user to directly assign shell program to execute the lines. It looks like a bare API calling a program.
How about adding an API like `os.shell(line)` to automatically detect the shell and run cmd line directly in xmake scripts?

**Reply by waruqi** [reply, marked as answer]
```lua
os.execv("bash", {"-c", "echo hello"})
```
